modules/ding_config_import/ding_config_import.module

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * @file
+ * Implementation of simple form to upload YAML file.
+ */
+
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Implements hook_menu().
+ */
+function ding_config_import_menu() {
+  $item = [];
+
+  $item['admin/config/ding/config'] = [
+    'title' => 'Upload configuration',
+    'description' => 'Upload and import configuration from YAML.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_config_import_admin_form'),
+    'access arguments' => array('administer site configuration'),
+  ];
+
+  return $item;
+}
+
+/**
+ * Administration form for yaml file upload.
+ */
+function ding_config_import_admin_form() {
+  $url = drupal_get_path('module', 'ding_config_import') . '/example.config.yml';
+
+  $form = [
+    '#attributes' => [
+      'enctype' => 'multipart/form-data',
+    ],
+    'wrapper' => [
+      '#type' => 'fieldset',
+      '#title' => t('Configuration upload'),
+      '#description' => t('This module comes with an <a href="/@url">example file</a> that can be used as an template for the upload. Note that enabled/disable modules may timeout, if it takes longer than the web-servers configuration allows.', ['@url' => $url]),
+    ],
+  ];
+
+  $form['wrapper']['upload'] = [
+    '#title' => t('Upload configurations file (yml)'),
+    '#type'  => 'file',
+  ];
+
+  $form['submit_upload'] = array(
+    '#type' => 'submit',
+    '#value' => t('Upload'),
+  );
+
+  return $form;
+}
+
+/**
+ * Form submit handler.
+ */
+function ding_config_import_admin_form_submit($form, &$form_state) {
+  $error = false;
+  $msg = t('The configuration options have been saved. You may need to clear cache to rebuild things if you have enabled/disabled modules etc.');
```

Comment:
This is somewhat vague 😅 . We could consider just always clearing the cache after an import. However this might increase the risk of timeouts - especially without Batch API.

Reply:
yes, therefor an improvement for later.